plugins/status.js

```diff
@@ -0,0 +1,245 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const async = require('async');
+
+exports.register = function () {
+    this.outbound = require('./../outbound');
+    this.queue_dir = require('./../outbound/queue').queue_dir;
+}
+
+exports.hook_capabilities = function (next, connection) {
+
+    if (connection.remote.ip === '127.0.0.1' || connection.remote.ip === '::1') {
+        connection.capabilities.push('STATUS');
+    }
+
+    next();
+}
+
+exports.hook_unrecognized_command = function (next, connection, params) {
+    const self = this;
+
+    const result_process = function (err, result) {
+        if (err) return next(DENY, err.message);
+        return connection.respond(211, result ? JSON.stringify(result) : "null", () => next(OK));
+    };
+
```

Comment:
This might be a dumb question. You've got 3 `if` conditions here and only two of them call the `next` callback. That doesn't "look" right. Is it?